rootfs/scheduler/__init__.py

```diff
@@ -993,6 +993,10 @@ def _wait_until_pods_are_ready(self, namespace, container, labels, desired):  #
             logger.info('timed out ({}s) waiting for pods to come up in namespace {}'.format(timeout, namespace))  # noqa
 
         logger.info("{} out of {} pods in namespace {} are in service".format(count, desired, namespace))  # noqa
+        if count != desired:
+            # raising to allow operations to rollback
+            raise KubeException('Not enough pods in namespace {} came into service. '
+                                '{} out of {}'.format(namespace, count, desired))
```

Comment:
That's a good point - Until we have both liveness and readiness as independently configurable then we will have to consider failure in either as a failure mode.

Even if we let people configure both then I think it is a bit of a problem if your readiness is still working but is beyond the timeout, we can't really start spinning working stuff down until we have enough pods to keep the app running. That's the biggest issue - Suppose we could add time to the timeout when we detect it is in an almost ready state, but only up to a point